project-base/tests/FrontendApiBundle/Functional/Product/PromotedProductsTest.php

```diff
@@ -0,0 +1,123 @@
+
+        self::assertEquals($productsExpected, $responseData, json_encode($responseData));
+    }
+
+    public function testPromotedProductsReturnsSameProductAsProductDetail(): void
+    {
+        $queryPromotedProducts = '
+            query {
+                promotedProducts {
+                    uuid
+                    name
+                    shortDescription
+                    link
+                    unit {
+                        name
+                    }
+                    isUsingStock
+                    availability {
+                        name
+                    }
+                    stockQuantity
+                    categories {
+                        name
+                    }
+                    flags {
+                        name
+                        rgbColor
+                    }
+                    price {
+                        priceWithVat
+                        priceWithoutVat
+                        vatAmount
+                    }
+                    brand {
+                        name
+                    }
+                }
+            }
+        ';
+
+        $graphQlType = 'promotedProducts';
+        $responsePromotedProducts = $this->getResponseContentForQuery($queryPromotedProducts);
+
+        $this->assertResponseContainsArrayOfDataForGraphQlType($responsePromotedProducts, $graphQlType);
+        $responseDataPromotedProducts = $this->getResponseDataForGraphQlType($responsePromotedProducts, $graphQlType);
+
+        $productUuid = $responseDataPromotedProducts[0]['uuid'];
+
+        $queryProductDetail = '
+            query {
+                product(uuid: "' . $productUuid . '") {
+                    uuid
+                    name
+                    shortDescription
+                    link
+                    unit {
+                        name
+                    }
+                    isUsingStock
+                    availability {
+                        name
+                    }
+                    stockQuantity
+                    categories {
+                        name
+                    }
+                    flags {
+                        name
+                        rgbColor
+                    }
+                    price {
+                        priceWithVat
+                        priceWithoutVat
+                        vatAmount
+                    }
+                    brand {
+                        name
+                    }
+                }
+            }
+        ';
+
+        $graphQlType = 'product';
+        $responseProductDetail = $this->getResponseContentForQuery($queryProductDetail);
+
+        $this->assertResponseContainsArrayOfDataForGraphQlType($responseProductDetail, $graphQlType);
+        $responseDataProductDetail = $this->getResponseDataForGraphQlType($responseProductDetail, $graphQlType);
+
+        self::assertEquals($responseDataPromotedProducts[0], $responseDataProductDetail);
```

Comment:
This can lead to undefined offset error. The failed test sounds more reasonable to me in that case.

Reply:
thanks, fixed in 6b185d6fa